test/sync/test.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/iotexproject/iotex-antenna-go/v2/iotex"
+	"github.com/iotexproject/iotex-core/blockchain/genesis"
+	"github.com/iotexproject/iotex-core/pkg/probe"
+	"github.com/iotexproject/iotex-core/server/itx"
+	"github.com/iotexproject/iotex-proto/golang/iotexapi"
+	_ "go.uber.org/automaxprocs"
+	"go.uber.org/zap"
+	glog "log"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/iotexproject/iotex-core/config"
+	"github.com/iotexproject/iotex-core/pkg/log"
+)
+
+func init() {
+	flag.Usage = func() {
+		_, _ = fmt.Fprintf(os.Stderr,
+			"usage: server -config-path=[string]\n")
+		flag.PrintDefaults()
+		os.Exit(2)
+	}
+	flag.Parse()
+}
+
+func main() {
+	go listenBlockSync()
+	startServer()
+}
+
+func listenBlockSync() {
+	// wait server started
+	time.Sleep(time.Minute * 5)
+	conn, err := iotex.NewDefaultGRPCConn("localhost:14014")
+	if err != nil {
+		glog.Fatal(err)
+	}
+	defer conn.Close()
+
+	c := iotex.NewReadOnlyClient(iotexapi.NewAPIServiceClient(conn))
+	meta, err := c.API().GetChainMeta(context.Background(), &iotexapi.GetChainMetaRequest{})
+	if err != nil {
+		glog.Fatal(err)
+	}
+	start := meta.ChainMeta.Height
+
+	for {
+		time.Sleep(time.Minute * 5)
+		meta, err = c.API().GetChainMeta(context.Background(), &iotexapi.GetChainMetaRequest{})
+		if err != nil {
+			glog.Fatal(err)
+		}
+		if start >= meta.ChainMeta.Height {
```

Comment:
start is not updated in the for loop. Therefore, if the first time it works, the if condition will always be true.

Reply:
```
	start := 0
	for {
		time.Sleep(time.Minute * 5)
		meta, err = c.API().GetChainMeta(context.Background(), &iotexapi.GetChainMetaRequest{})
		if err != nil {
			log.L().Fatal(err)
		}
		if start >= meta.ChainMeta.Height {
			glog.Fatalf("blockchain stop sync at %d height", start)
		}
                start = meta.ChainMeta.Height
	}
```